omnigibson/utils/usd_utils.py

```diff
@@ -682,25 +685,18 @@ def create_mesh_prim_with_default_xform(primitive_type, prim_path, u_patches=Non
     hs_backup = carb.settings.get_settings().get(evaluator.SETTING_OBJECT_HALF_SCALE)
     carb.settings.get_settings().set(evaluator.SETTING_U_SCALE, 1)
     carb.settings.get_settings().set(evaluator.SETTING_V_SCALE, 1)
+    stage = og.sim.stage if stage is None else stage
 
     # Default half_scale (i.e. half-extent, half_height, radius) is 1.
     # TODO (eric): change it to 0.5 once the mesh generator API accepts floating-number HALF_SCALE
     #  (currently it only accepts integer-number and floors 0.5 into 0).
     carb.settings.get_settings().set(evaluator.SETTING_OBJECT_HALF_SCALE, 1)
+    kwargs = dict(prim_type=primitive_type, prim_path=prim_path, stage=stage)
     if u_patches is not None and v_patches is not None:
-        omni.kit.commands.execute(
-            "CreateMeshPrimWithDefaultXform",
-            prim_type=primitive_type,
-            prim_path=prim_path,
-            u_patches=u_patches,
-            v_patches=v_patches,
-        )
-    else:
-        omni.kit.commands.execute(
-            "CreateMeshPrimWithDefaultXform",
-            prim_type=primitive_type,
-            prim_path=prim_path,
-        )
+        kwargs["u_patches"] = u_patches
+        kwargs["v_patches"] = v_patches
+
+    CreateMeshPrimWithDefaultXformCommand(**kwargs).do()
```

Comment:
interesting syntax, what is the benefit of using this new API?

Reply:
this (and all of the associated boilerplate code) is to allow us to leverage `create_primitive_mesh` and specify a stage explicitly. Omni's default code automatically uses the active stage, but for our post-processing pipeline where we now want to generate sphere meshes for the toggle button we should explicitly pass in the stage opened for a given object asset

Reply:
I see. For the post-processing pipeline, why is the default active stage not sufficient?

Reply:
you're right -- I didn't think about that. I was thinking about our add_default_metalinks pipeline, which actually has multiple stages open at once, and the active one (the OG one) is not the one we actually want to add metalinks to.

Reply:
Sorry I am not sure I understand your response. So do we have the specify the stage, or is the default stage sufficient?

Reply:
From what you said, it seems the default active stage is NOT sufficient?

Reply:
sorry for not being clear. what i mean is:

- in the `generate_metadata` script in `ig_pipeline`, I think the default stage is fine
- in the `generate_default_metalinks` script in OG, the default stage is **not** fine

So I think the change actually wasn't technically needed, but it's good to be consistent in case we need to generate additional mesh-based metalinks in the future not in the ig_pipeline